src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,203 @@
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_negativeUserAge_NotOk() {
+        testUser.setAge(-20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_noAcceptableUserAge_NotOk() {
+        testUser.setAge(10);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_sameUserLoginRegistered_NotOk() throws CloneNotSupportedException {
+        String repeatingLogin = "same_login_user";
+        testUser.setLogin(repeatingLogin);
+        Storage.people.add(testUser);
+        User sameLoginUser = new User();
+        sameLoginUser.setId(243568765L);
+        sameLoginUser.setLogin(repeatingLogin);
+        sameLoginUser.setPassword("some_valid_password");
+        sameLoginUser.setAge(31);
+
+        Storage.people.add(testUser);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_correctLengthUserLogin_Ok() throws RegistrationException {
+        testUser.setLogin("normal_length_login");
+        registrationService.register(testUser);
+        assertTrue(testUser.getLogin().length() >= MIN_LOGIN_LENGTH);
+    }
+
+    @Test
+    void register_minLengthUserLogin_Ok() throws RegistrationException {
+        testUser.setLogin("normal");
+        registrationService.register(testUser);
+        assertTrue(testUser.getLogin().length() == MIN_LOGIN_LENGTH);
+    }
+
+    @Test
+    void register_correctLengthUserPassword_Ok() throws RegistrationException {
+        testUser.setPassword("normal_length_password");
+        registrationService.register(testUser);
+        assertTrue(testUser.getPassword().length() >= MIN_PASSWORD_LENGTH);
+    }
+
+    @Test
+    void register_minLengthUserPassword_Ok() throws RegistrationException {
+        testUser.setPassword("passwd");
+        registrationService.register(testUser);
+        assertTrue(testUser.getPassword().length() == MIN_PASSWORD_LENGTH);
+    }
+
+    @Test
+    void register_minUserAge_Ok() throws RegistrationException {
+        testUser.setAge(18);
+        registrationService.register(testUser);
+        assertTrue(testUser.getAge() == MIN_USER_ACCEPTABLE_AGE);
+    }
+
+    @Test
+    void register_acceptableUserAge_Ok() throws RegistrationException {
+        testUser.setAge(28);
+        registrationService.register(testUser);
+        assertTrue(testUser.getAge() >= MIN_USER_ACCEPTABLE_AGE);
+    }
+
+    @Test
+    void register_validUserRegistration_Ok()
+            throws RegistrationException, CloneNotSupportedException {
+        assertTrue(Storage.people.isEmpty());
+        registrationService.register(testUser);
+        assertTrue(Storage.people.size() == 1);
```

Comment:
asserts should be in the end of test method

Reply:
changed